src/osd/CMakeLists.txt

```diff
@@ -2,6 +2,13 @@ set(osdc_osd_srcs
   ${CMAKE_SOURCE_DIR}/src/osdc/Objecter.cc
   ${CMAKE_SOURCE_DIR}/src/osdc/Striper.cc)
 
+if(WITH_OSD_INSTRUMENT_FUNCTIONS)
+  set(GCC_C_FLAGS "-finstrument-functions")
+  set(GCC_C_FLAGS "${GCC_C_FLAGS} -finstrument-functions-exclude-function-list=_mm_loadu_si128,_mm_cmpeq_epi32,_mm_movemask_epi8")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GCC_C_FLAGS}")
```

Comment:
The -exclude part is to resolve a link-time issue. The functions listed in the -exclude part are inline, but GCC tries to instrument them as well. I updated the code and documentation of this PR to support WITH_OSD_INSTRUMENT_FUNCTIONS on Clang, but with no effect. Clang might eventually support the -exclude option (there are patches for it online).